### PR TITLE
Fix images docs paths and improves auto typing

### PR DESCRIPTION
**Pull request by edgarriba**
### Description

- Fixes the links to the auto-generated images in kornia.augmentation docs.
- Fixes an issue with the `sphinx_rtd_theme` by upgrading the version.
- Resizes images to improve efficiency to load website.
- Adds `RandomMixUp` and `RandomCutMix` images to docs.
- Includes the following sphinx plugins
  -  `sphinx_autodoc_typehints`: automatically include the typings in the docstrings. In case typing is added by hand, it does nothing.
  - `sphinx_copybutton`: add a copy button next to the docstring snippets.
  - `sphinx_autodoc_defaultargs`: adds to the end of each docstrings arguments the default value.
- Updates `kornia.[augmentation, color]` docstrings.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Comment by edgarriba** [comment]
@shijianjian seems that it worked ! I think that we should find in some case more extreme parameters to show clearly what's happening.

File: docs/generate_example_images.py [hunk 1]
```diff
@@ -8,62 +9,72 @@
...
-    "RandomSolarize": (0.1,),
-    "RandomVerticalFlip": (),
-    "RandomThinPlateSpline": (),
-}
+    # convert to image array
+    img: np.ndarray = cv2.imdecode(nparr, cv2.IMREAD_UNCHANGED)
 
-# ITERATE OVER THE TRANSFORMS
+    # convert the image to a tensor
+    img_t: torch.Tensor = K.utils.image_to_tensor(img, keepdim=False)  # 1xCxHXW
```
Comment (shijianjian) [inline docs/generate_example_images.py:32]: A quick note to resize the image first. Otherwise, it gonna be too heavy for webpages.

File: docs/generate_example_images.py [hunk 2]
```diff
@@ -8,62 +9,72 @@
...
-    out = torch.cat([img_t, out], dim=-1)
-    # save the output image
-    out_np = K.utils.tensor_to_image((out * 255.0).byte())
-    cv2.imwrite(str(OUTPUT_PATH / f"{aug_name}.jpg"), out_np)
+    # TODO: make this more generic for modules out of kornia.augmentation
+    # Dictionary containing the transforms to generate the sample images:
+    # Key: Name of the transform class.
+    # Value: The default parameters to use.
+    augmentations_list: dict = {
+        "ColorJitter": (0.1, 0.1, 0.1, 0.1),
```
Comment (shijianjian) [inline docs/generate_example_images.py:41]: Probably add a seed for each transform to be deterministic.

File: docs/source/conf.py [hunk 3]
```diff
@@ -3,6 +3,14 @@
 
 import sphinx_rtd_theme
 
+import importlib.util
+spec = importlib.util.spec_from_file_location("generate_example_images", "../generate_example_images.py")
```
Comment (shijianjian) [inline docs/source/conf.py:7]: Add some comments here to indicate that the doc build happens in an isolated container, so that we need to include the preprocessing part manually here.

**Review by shijianjian — COMMENTED** [review]
Great. Let's do the following.

File: docs/generate_example_images.py [hunk 4]
```diff
@@ -27,53 +27,57 @@ def main():
...
     }
 
     # ITERATE OVER THE TRANSFORMS
 
-    for aug_name, args in augmentations_list.items():
+    for aug_name, (args, num_samples, seed) in augmentations_list.items():
+        img_in = img_t.repeat(num_samples, 1, 1, 1)
         # dynamically create the class instance
         cls = getattr(mod, aug_name)
         aug = cls(*args, p=1.0)
```
Comment (edgarriba) [inline docs/generate_example_images.py:71]: @shijianjian the seed could be added directly here I guess

Comment (shijianjian) [reply docs/generate_example_images.py:71]: We have had a discussion at some point about adding a seed param to the function. While it means to maintain multiple sub RNG states in order to not mess up the main thread. Not an easy add-on, I will therefore evaluate further with the current API.

File: kornia/color/lab.py [hunk 5]
```diff
@@ -15,14 +15,16 @@
 def rgb_to_lab(image: torch.Tensor) -> torch.Tensor:
     r"""Converts a RGB image to Lab.
 
+    .. image:: _static/img/rgb_to_lab.png
```
Comment (edgarriba) [inline kornia/color/lab.py:18]: @cceyda @ducha-aiki @shijianjian @lferraz how should we represent this ? by default we get now something like this
image

Comment (cceyda) [reply kornia/color/lab.py:18]: showing any image not encoded in RGB makes no sense...
So maybe we can show a single pixel value.  
rgb(before conversion) & lab (after conversion) input[0,:,0,0] vs output[0,:,0,0] 
something like this:
image
